internal/controller/controllers/agent_controller.go

```diff
@@ -437,14 +437,14 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 			if err != nil {
 				return ctrl.Result{Requeue: true}, nil
 			}
-		}
-		if agent.Status.DebugInfo.LogsURL == "" && !time.Time(h.LogsCollectedAt).Equal(time.Time{}) { // logs collection time is updated means logs are available
-			logsURL, err := generateControllerLogsDownloadURL(r.ServiceBaseURL, clusterId.String(), r.AuthType, agent.Name, "host")
-			if err != nil {
-				log.WithError(err).Error("failed to generate controller logs URL")
-				return ctrl.Result{}, err
+			if agent.Status.DebugInfo.LogsURL == "" && !time.Time(h.LogsCollectedAt).Equal(time.Time{}) { // logs collection time is updated means logs are available
```

Comment:
What will happen to the url in case the host is bound to a new cluster? 
It looks like we will keep the same URL without the new cluster no? 
it may involve UX as well, i see two options 
1. keep the old url until the host get bound to a new cluster - can help debug issues in case host was replaced for some reason but may be confusing to uses that we will see logs on an unbound host
2. Delete the url when host is unbound - we lose the logs but it will not confuse the user
Crystal what do you think?

Reply:
I would expect that whatever is responsible for binding (or maybe unbinding) a host would clear these fields.
There should be enough time to grab the url for a failed host before it gets reassigned.

That all said, I think this PR can fix the bug first as we were not clearing the logs (or events for that matter) before this change, right?

Reply:
Sounds good to me. 
Regardless need to cover it with a unit tests https://github.com/openshift/assisted-service/pull/3365#pullrequestreview-884949401

Reply:
Good point about when a host becomes unbound! 

I agree, I think we can create a separate PR/issue for investigating the log removal when a host is unbound. I'll take a look at the current unbound logic and see if we can include it there.

Reply:
> A question I have is: internally, when a host is unbound, do the logs get removed from storage?
I don't think so 
The cleanup usually done when deleting the cluster.